File: demos/03-cosmos-db/demo-03/food-orders-api/Common/Order.Model.cs
using Newtonsoft.Json;

namespace FoodApp.Orders
{
    public class Order
    {
        [JsonProperty(PropertyName = "id")]
        public string Id { get; set; }
        [JsonProperty(PropertyName = "customerId")]
        public int CustomerId { get; set; }
        [JsonProperty(PropertyName = "type")]
        public string Type { get; set; }
        [JsonProperty(PropertyName = "total")]
        public decimal Total { get; set; }
        [JsonProperty(PropertyName = "customer")]
        public Customer Customer { get; set; }
        [JsonProperty(PropertyName = "items")]
        public List<OrderItem> Items { get; set; }
    }

    public class Customer
    {
        [JsonProperty(PropertyName = "name")]
        public string Name { get; set; }
        [JsonProperty(PropertyName = "email")]
        public string EMail { get; set; }
        [JsonProperty(PropertyName = "address")]
        public string Address { get; set; }
    }

    public class OrderItem
    {
        [JsonProperty(PropertyName = "id")]
        public int Id { get; set; }
        [JsonProperty(PropertyName = "name")]
        public string Name { get; set; }
        [JsonProperty(PropertyName = "price")]
        public decimal Price { get; set; }
        [JsonProperty(PropertyName = "quantity")]
        public int Quantity { get; set; }
    }

    public enum OrderStatus
    {
        cart,
        placed,
        paid,
        preparing,
        ready_for_delivery,
        delivered,
        rejcted
    }
}

File: demos/05-containers/demo-01/food-shop/src/app/shop/order/order.model.ts
import { CartItem } from '../cart-item.model';
export class Order {
  constructor() {
    this.customer = new Customer();
    this.payment = new Payment();
    this.shippingAddress = new Address();
    this.items = [];
  }

  id?: string;
  type = 'order'
  customer: Customer;
  shippingAddress: Address;
  payment: Payment;
  items: CartItem[];
  total = 0;
}

export class Payment {
  type = '';
  accountNumber = '';
}

export class Address {
  street = '';
  city = '';
  country = '';
  zipCode = '';
}

export class Customer {
  id = '';
  name = '';
  email = '';
  phone = '';
}